### ISSUE TITLE
模型效果

### ISSUE THREAD

**Issue by prettyprettyboy** [Y-debug-sys/Diffusion-TS#1]
模型无条件生成的结果是被保存为ddpm_fake.npy吗？为什么我构造的一些简单的sin曲线数据，生成出来的效果画出来很差呢？

**Comment by Y-debug-sys** [comment]
Hi, can you provide me with more information on the running scripts and your synthetic curve? I cannot reproduce this problem. Thanks a lot.

**Comment by Y-debug-sys** [comment]
First, I hope you can show me a comparison between the training data and generated data (just one example is OK). Additionally, for long or frequently occurring sequences, I suggest enlarging the receptive field for 'season' part in the config (adjusting kernel_size and padding).

**Comment by prettyprettyboy** [comment]
我直接在stocks上测试，然后对比ddpm_fake_stocks.npy和对应的stocks_ground_truth_train.npy同一维度的绘制出来的结果都不太一样

**Comment by Y-debug-sys** [comment]
The stocks data set is very different before and after (you can print it to see) and there is no direct connection between the unconditional generation and a random real sample. I suggest you use some distribution-related metrics, or find similar segments in the training set to visualize for comparison. And can you show me your plotting in the first question?

**Comment by prettyprettyboy** [comment]
![Figure_1](https://github.com/Y-debug-sys/Diffusion-TS/assets/72538202/c2064818-8561-4dad-b2ba-34bb22b1c92a)
![Figure_2](https://github.com/Y-debug-sys/Diffusion-TS/assets/72538202/a6b1c371-e204-49b1-8e2d-cf354553e0f5)
我没改代码测试了stocks和etth数据的无条件生成，选择其中一维变量画出来，如上图所示，第一张是原始数据，第二张是生成数据，这几个数据集生成的序列基本上都很像这个第二张图，点都是平的线，没有起伏

**Comment by prettyprettyboy** [comment]
我自己用np.sin(np.linespace(0,10,300))生成了一些sin函数的单变量序列训练后做无条件生成，生成的序列也很像上述第二幅图。也就是序列只在上面一排和下面一排有数据，没有周期和起伏

**Comment by Y-debug-sys** [comment]
Sorry for that, please cancel the comment **"trainer.load(args.milestone)"**  in main.py. I forget it after debug.

**Comment by Y-debug-sys** [comment]
Sorry for that, please cancel the comment "trainer.load(args.milestone)" in main.py. I forget it before my submit.

**Comment by prettyprettyboy** [comment]
解决了，感谢！还有个问题是，你之前说的长的周期性序列要调整kernel_size 和 padding，具体怎么调整呢，比如大概增加多少，还是说完全根据实验结果来试，目前我自己的数据长度是700、800的样子

**Comment by Y-debug-sys** [comment]
The model is designed for experiments with a length not exceeding 300, and the features of TS datasets may be very different. Therefore, I cannot provide a specific value pair.

**Comment by Y-debug-sys** [comment]
The model is designed for experiments with a length not exceeding 300, and the features of TS datasets are usually different.   Therefore, I cannot provide a specific value pair. Some optimizations on your data may be inevitable.
